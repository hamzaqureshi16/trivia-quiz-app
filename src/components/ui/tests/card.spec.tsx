import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Card } from '../card';

describe('Card Component', () => {
  it('renders the Card with children', () => {
    render(
      <Card>
        <p>Card Content</p>
      </Card>,
    );
    expect(screen.getByText('Card Content')).toBeInTheDocument();
  });

  it('applies custom class names', () => {
    render(
      <Card className="custom-class">
        <p>Styled Card</p>
      </Card>,
    );
    const cardElement = screen.getByText('Styled Card').parentElement;
    expect(cardElement).toHaveClass('custom-class');
  });

  it('matches the snapshot', () => {
    const { container } = render(
      <Card>
        <p>Snapshot Card</p>
      </Card>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
