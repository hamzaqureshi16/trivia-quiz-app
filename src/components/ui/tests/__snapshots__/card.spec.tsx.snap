// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches the snapshot 1`] = `
<div
  class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
  data-slot="card"
>
  <p>
    Snapshot Card
  </p>
</div>
`;
